core/src/main/java/com/novoda/noplayer/internal/exoplayer/NoPlayerAdsLoader.java

```diff
@@ -150,6 +159,10 @@ public void release() {
     public void handlePrepareError(int adGroupIndex, int adIndexInAdGroup, IOException exception) {
         if (adPlaybackState != null) {
             adPlaybackState = adPlaybackState.withAdLoadError(adGroupIndex, adIndexInAdGroup);
+
+            AdvertBreak advertBreak = advertBreaks.get(adGroupIndex);
+            Advert advert = advertBreak.adverts().get(adIndexInAdGroup);
+            advertListener.onAdvertPrepareError(advert, exception);
```

Comment:
Pass back as much information as possible.

Reply:
Should never be passed `-1` or incorrect values as it is directly correlated to the `AdPlaybackState` which in turn is created from our models.

Reply:
We receive all of these on the `main-thread` so we don't need to worry about using a handler here.

Reply:
I guess that we assume that those gets are not going to return nulls or throw exceptions...

Reply:
They can't return `null`, if it contained a `null` for an `advertBreak` then it would fail in another place when loading the adverts. As for throwing, it shouldn't throw, these indexes are defined by the source which comes from our playback state, which again, comes from our models.